### ISSUE TITLE
Remove rkt container runtime

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#3906]
While the `rkt` component of kubernetes that was added in 1.3 and removed in 1.9 is gone,
there is a CRI replacement called `rktlet`. But even that is also stuck in the 1.8 - 1.11 era ?

So it seems like there is not too much development going on with `rkt`, and nobody using it.
Users looking for a supported alternative container runtime would be better off with `cri-o`...

Therefore the suggestion is to **remove** it, for 1.0:

https://github.com/kubernetes/minikube/compare/master...afbjorklund:rkt-remove

I made a small effort to update it, in case I am wrong:

https://github.com/kubernetes/minikube/compare/master...afbjorklund:rktlet

**Comment by afbjorklund** [comment]
Reference #168
> I managed to install rkt in the initrd image (grows from to 37MB to 75MB, do you foresee a problem with this increase?

And #571 (for ISO)
> I managed to build a ~57MB big bootable systemd-as-pid1-based ISO image having rkt included using buildroot.

Ping @tstromberg

**Comment by tstromberg** [comment]
Let's nuke it.

**Comment by tstromberg** [comment]
Related: #3920

**Comment by tstromberg** [comment]
One quirk: the `toolbox` binary currently references on rkt. I'm not sure yet if it's truly required, or if there is a viable rkt-less workaround.

**Comment by afbjorklund** [comment]
@tstromberg : yeah, I saw that. Maybe upstream has some kind of workaround for it, will peek around.

https://github.com/coreos/toolbox

> There are currently two scripts that live within this repository:
> 
> toolbox: designed for Container Linux, uses rkt and systemd-nspawn
> rhcos-toolbox: designed for Red Hat CoreOS, uses podman

**Comment by afbjorklund** [comment]
Deleted the toolbox for now, seems like _running_ containers with podman isn't working at the moment.

``` console
                         _             _            
            _         _ ( )           ( )           
  ___ ___  (_)  ___  (_)| |/')  _   _ | |_      __  
/' _ ` _ `\| |/' _ `\| || , <  ( ) ( )| '_`\  /'__`\
| ( ) ( ) || || ( ) || || |\`\ | (_) || |_) )(  ___/
(_) (_) (_)(_)(_) (_)(_)(_) (_)`\___/'(_,__/'`\____)
$ sudo podman pull busybox
Trying to pull docker.io/busybox:latest...Getting image source signatures
Copying blob 697743189b6d: 738.18 KiB / 738.18 KiB [========================] 1s
Copying config d8233ab899d4: 1.46 KiB / 1.46 KiB [==========================] 0s
Writing manifest to image destination
Storing signatures
d8233ab899d419c58cf3634c0df54ff5d8acc28f8173f09c21df4a07229e1205
$ sudo podman run -it busybox
error configuring network namespace for container b731699b47263152e29888a9b91d671346548c0d059ec056a8aa8ce1797077d6: Missing CNI default network
```

Probably needs some CNI configuration.

**Comment by afbjorklund** [comment]
Found a way to use podman for the extraction, but still use systemd-nspawn for running.

**Comment by tstromberg** [comment]
Can this be closed?

**Comment by afbjorklund** [comment]
Closed in 1.0.
